code/datums/mood.dm

```diff
@@ -0,0 +1,493 @@
+#define MINOR_INSANITY_PEN 5
+#define MAJOR_INSANITY_PEN 10
+#define MOOD_CATEGORY_NUTRITION "nutrition"
+#define MOOD_CATEGORY_AREA_BEAUTY "area_beauty"
+
+/**
+ * Mood datum
+ *
+ * Contains the logic for controlling a living mob's mood and sanity.
+ */
+/datum/mood
+	/// Weakref to the parent (living) mob
+	var/datum/weakref/parent
+
+	/// The total combined value of all moodlets for the mob
+	var/mood
+	/// Current sanity of the mob (ranges from 0 - 150)
+	var/sanity = SANITY_NEUTRAL
+	/// the total combined value of all visible moodlets for the mob
+	var/shown_mood
+	/// Moodlet value modifier
+	var/mood_modifier = 1
+	/// Used to track what stage of moodies they're on (1-9)
+	var/mood_level = MOOD_LEVEL_NEUTRAL
+	/// To track what stage of sanity they're on (1-6)
+	var/sanity_level = SANITY_LEVEL_NEUTRAL
+	/// Is the owner being punished for low mood? if so, how much?
+	var/insanity_effect = 0
+	/// The screen object for the current mood level
+	var/atom/movable/screen/mood/mood_screen_object
+
+	/// List of mood events currently active on this datum
+	var/list/mood_events = list()
+
+/datum/mood/New(mob/living/mob_to_make_moody)
+	if (!istype(mob_to_make_moody))
+		stack_trace("Tried to apply mood to a non-living atom!")
+		qdel(src)
+		return
+
+	START_PROCESSING(SSmood, src)
+
+	parent = WEAKREF(mob_to_make_moody)
+
+	RegisterSignal(mob_to_make_moody, COMSIG_MOB_HUD_CREATED, .proc/modify_hud)
+	RegisterSignal(mob_to_make_moody, COMSIG_ENTER_AREA, .proc/check_area_mood)
+	RegisterSignal(mob_to_make_moody, COMSIG_LIVING_REVIVE, .proc/on_revive)
+	RegisterSignal(mob_to_make_moody, COMSIG_MOB_STATCHANGE, .proc/handle_mob_death)
+
+	mob_to_make_moody.become_area_sensitive(MOOD_DATUM_TRAIT)
+	if(mob_to_make_moody.hud_used)
+		modify_hud()
+		var/datum/hud/hud = mob_to_make_moody.hud_used
+		hud.show_hud(hud.hud_version)
+
+/datum/mood/Destroy(force, ...)
+	STOP_PROCESSING(SSmood, src)
+
+	QDEL_LIST_ASSOC_VAL(mood_events)
+
+	unmodify_hud()
+	var/mob/living/mob_parent = parent?.resolve()
+	if (mob_parent)
+		mob_parent.lose_area_sensitivity(MOOD_DATUM_TRAIT)
+		UnregisterSignal(mob_parent, list(COMSIG_MOB_HUD_CREATED, COMSIG_ENTER_AREA, COMSIG_LIVING_REVIVE, COMSIG_MOB_STATCHANGE))
+
+	return ..()
+
+/datum/mood/process(delta_time)
+	var/mob/living/mob_parent = parent?.resolve()
```

Comment:
?. is just a habit i got into i think all the cases it would be null are accounted for.

I used a weakref because generally thats the pattern i've been told
parent holds a ref to child
child uses a weakref to parent
is the safest way, but i could definitely just do it with a deletion signal